lib/src/main/scala/spinal/lib/sim/bus/wishbone/WishboneDriver.scala

```diff
@@ -7,30 +7,60 @@ import spinal.lib.bus.wishbone._
 import scala.collection.immutable._
 import scala.util.Random
 
-object WishboneDriver{
-  def apply(bus: Wishbone, clockdomain: ClockDomain) = new WishboneDriver(bus,clockdomain)
+object WishboneDriver {
+  def apply(bus: Wishbone, clockdomain: ClockDomain) = new WishboneDriver(bus, clockdomain)
+  def apply(bus: Wishbone) = new WishboneDriver(bus, bus.CYC.clockDomain)
 }
 
 /** This is a helping class for driving the wishbone bus
   * @param bus the wishbone bus to drive
   * @param clockdomain the clockdomain where the bus reside
   */
-class WishboneDriver(bus: Wishbone, clockdomain: ClockDomain){
+class WishboneDriver(val bus: Wishbone, clockdomain: ClockDomain){
   val busStatus = WishboneStatus(bus)
 
+  private def initAsMaster(): this.type = {
+    bus.CYC #= false
+    bus.STB #= false
+    bus.WE #= false
+    bus.ADR #= 0
+    bus.DAT_MOSI #= 0
+    Option(bus.LOCK).foreach(_ #= false)
+    Option(bus.SEL).foreach(_ #= 0)
+    this
+  }
+
+  private def initAsSlave() : this.type = {
+    Option(bus.STALL).foreach(_ #= true)
+    Option(bus.RTY).foreach(_ #= false)
+    Option(bus.ERR).foreach(_ #= false)
+
+    bus.ACK #= false
+    bus.DAT_MISO #= 0
+    this
+  }
+
+  def init() : this.type = {
+    if(!bus.isMasterInterface)
+      initAsMaster()
+    else
+      initAsSlave()
+  }
+
+  init()
+
   /** Drive the wishbone bus as master with a transaction.
     * @param transaction The transaction to send.
     */
-  def sendAsMaster(transaction : WishboneTransaction, we: Boolean): Unit = {
+  private def sendAsMaster(transaction : WishboneTransaction, we: Boolean): Unit = {
```

Comment:
We shouldn't make functions private if we don't want to break users of the standard mode functionality

Reply:
I can reverse this; but I think the intent of the class is that everything uses `send` and it delegates out to the other functions. I'm not sure if anyone should be calling sendAsMaster directly? I guess the bus can be directionless in the sim but I'm not sure I see a use case for that.

But yeah, I guess since it can break existing code I'll reverse it. Maybe tagged as deprecated?